manifests/init.pp

```diff
@@ -81,6 +83,7 @@
 
   Boolean $reverse_proxy = false,
   Stdlib::Port $reverse_proxy_port = 8443,
+  Boolean $reverse_proxy_use_http2_to_backend = true,
```

Comment:
I wonder if
```puppet
Enum['h2','https'] $backend_protocol = 'h2'
```
is cleaner?

Reply:
Not sure about the answer. On the one hand, it feels like an internal implementation detail which protocol we use. On the other hand, I can see some flexibility here. Other options are (in theory) h2c (HTTP/2 cleartext) and possibly in the future HTTP/3.

Reply:
Isn't `reverse_proxy_use_http2_to_backend` leaking the same detail? ;)
But yeah, the flexibility to add H3 in the future was what I was aiming at.

Wouldn't block on this, but wanted to open up the discussion.

Reply:
Updated